controllers/resources/lease.go

```diff
@@ -0,0 +1,233 @@
+	ObtainNodeLease(remediationCR *unstructured.Unstructured, nhc *remediationv1alpha1.NodeHealthCheck) (bool, *time.Duration, error)
+	//ManageLease extends or releases a lease based on the CR status, type of remediation and how long the lease is already leased
+	ManageLease(ctx context.Context, remediationCR *unstructured.Unstructured, nhc *remediationv1alpha1.NodeHealthCheck) (time.Duration, error)
+	// InvalidateLease extends or releases a lease based on the CR status, type of remediation and how long the lease is already leased
+	InvalidateLease(ctx context.Context, remediationCR *unstructured.Unstructured) error
+}
+
+type nhcLeaseManager struct {
+	client             client.Client
+	commonLeaseManager lease.Manager
+	log                logr.Logger
+}
+
+func NewLeaseManager(client client.Client, log logr.Logger) (LeaseManager, error) {
+	newManager, err := lease.NewManager(client, holderIdentity)
+	if err != nil {
+		log.Error(err, "couldn't initialize lease manager")
+		return nil, err
+	}
+	return &nhcLeaseManager{
+		client:             client,
+		commonLeaseManager: newManager,
+		log:                log.WithName("nhc lease manager"),
+	}, nil
+}
+
+func (m *nhcLeaseManager) ObtainNodeLease(remediationCR *unstructured.Unstructured, nhc *remediationv1alpha1.NodeHealthCheck) (bool, *time.Duration, error) {
+	nodeName := remediationCR.GetName()
+	leaseDuration := m.getLeaseDurationForRemediation(remediationCR, nhc)
+	leaseDurationWithBuffer := leaseDuration + LeaseBuffer
+
+	node := &v1.Node{}
+	if err := m.client.Get(context.Background(), types.NamespacedName{Name: nodeName}, node); err != nil {
+		m.log.Error(err, "couldn't obtain node lease node error getting node", "node name", nodeName)
+		return false, nil, err
+	}
+
+	if err := m.commonLeaseManager.RequestLease(context.Background(), node, leaseDurationWithBuffer); err != nil {
+		if _, ok := err.(lease.AlreadyHeldError); ok {
+			m.log.Info("can't acquire node lease, it is already owned by another owner", "already held error", err)
+			return false, &RequeueIfLeaseTaken, err
+		}
+
+		m.log.Error(err, "couldn't obtain lease for node", "node name", nodeName)
+		return false, nil, err
+	}
+
+	//all good lease created with wanted duration
+	return true, &leaseDuration, nil
+
+}
+
+func (m *nhcLeaseManager) ManageLease(ctx context.Context, remediationCR *unstructured.Unstructured, nhc *remediationv1alpha1.NodeHealthCheck) (time.Duration, error) {
+	node := &v1.Node{}
+	if err := m.client.Get(ctx, client.ObjectKey{Name: remediationCR.GetName()}, node); err != nil {
+		m.log.Error(err, "managing lease - couldn't fetch node", "node name", remediationCR.GetName())
+		return 0, err
+	}
+	l, err := m.commonLeaseManager.GetLease(ctx, node)
+	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
>e.g. get a new one

why ? in case lease doesn't exist it would be because it's deleted because it's overdue

Reply:
anyone can delete the lease 🤷🏼‍♂️ 🙂

Reply:
IIUC we can abandon this thread, since we are already discussing this in the [thread below](https://github.com/medik8s/node-healthcheck-operator/pull/212#discussion_r1267886016.)